server/src/main/java/io/crate/expression/scalar/AgeFunction.java

```diff
@@ -0,0 +1,226 @@
+            module.register(
+                Signature.scalar(
+                    NAME,
+                    DataTypes.STRING.getTypeSignature(),
+                    dataType.getTypeSignature(),
+                    DataTypes.LONG.getTypeSignature()
+                ), AgeFunction::new
+            );
+
+            module.register(
+                Signature.scalar(
+                    NAME,
+                    dataType.getTypeSignature(),
+                    dataType.getTypeSignature(),
+                    DataTypes.LONG.getTypeSignature()
+                ),
+                AgeFunction::new
+            );
+            module.register(
+                Signature.scalar(
+                    NAME,
+                    DataTypes.STRING.getTypeSignature(),
+                    dataType.getTypeSignature(),
+                    dataType.getTypeSignature(),
+                    DataTypes.LONG.getTypeSignature()
+                ),
+                AgeFunction::new
+            );
+
+        }
+    }
+
+
+    public AgeFunction(Signature signature, Signature boundSignature)
+    {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+
+    @Override
+    public Long evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>... args)
+    {
+
+        assert args.length >= 1 &&  args.length <= 4 :
+            "Signature must ensure that there are  one,two ,three or four arguments";
+
+        if (checkNullInput(args))
+            return null;
+
+        Object earliestValue = args[args.length - 1].value();
+        Long timestampEarliest = DataTypes.TIMESTAMPZ.sanitizeValue(earliestValue);
+
+
+        Pair<String,Long> pair =   getTimezoneAndNewestDateValue(txnCtx, args);
+        Long timestampNewest= pair.getSecond();
+        String timeZoneString = pair.getFirst();
+        DateTimeZone timeZone = TimeZoneParser.parseTimeZone(timeZoneString);
+        DateTime dateNewest = new DateTime(timestampNewest.longValue(),timeZone);
+        DateTime dateEarliest = new DateTime(timestampEarliest.longValue(),timeZone);
+
+
+        Long duration = dateNewest.getMillis() - dateEarliest.getMillis();
+        return duration;
+    }
+
+
+    private boolean checkNullInput(Input<Object>[] args)
+    {
+        if(args.length==1)
+        {
+            if (args[0] == null)
+            {
+                return true;
+            }
+        }
+        else  if(args.length==2){
+            if (args[0] == null || args[1]==null)
+            {
+                return true;
+            }
+        }
+        else if(args.length==3){
+            if (args[0] == null || args[1]==null || args[2] ==null)
+            {
+                return true;
+            }
+        }
+        else  if(args.length==4 || args[1]==null || args[2]==null || args[3]==null){
+            if (args[0] == null)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
note: this comment considers that we don't support timezone, so can have only 1 or 2 argument
we probably can simplify null check (assuming that we did args length assertion before):
```
if (args[0] = null) {
    return null;
}
if ( args.length == 2 && args[1] == null) {
    return null;
}
```